src/lib/comms/sol-coap.c

```diff
@@ -678,11 +684,49 @@ timeout_expired(struct sol_coap_server *server, struct outgoing *outgoing)
     return expired;
 }
 
+static struct sol_buffer *prepare_buffer(struct outgoing *outgoing)
+{
+    static struct sol_buffer buffer;
+    struct sol_coap_packet *header = outgoing->header;
+    struct sol_coap_packet *payload = outgoing->pkt;
+    uint16_t new_size, new_offset, old_offset;
+    uint8_t new_tkl, old_tkl;
+    int r;
+
+    if (!header) {
+        buffer = SOL_BUFFER_INIT_CONST(payload->buf.data, payload->buf.used);
+        return &buffer;
```

Comment:
okay this is a PoC/Draft, but why return a pointer to the local buffer (even if static)? This smells problems, maybe just return the buffer (not a pointer to)?